mongo/errors.go

```diff
@@ -202,13 +202,14 @@ type ServerError interface {
 	HasErrorMessage(string) bool
 	// HasErrorCodeWithMessage returns true if any of the contained errors have the specified code and message.
 	HasErrorCodeWithMessage(int, string) bool
-
-	serverError()
 }
 
 var _ ServerError = CommandError{}
 var _ ServerError = WriteException{}
 var _ ServerError = BulkWriteException{}
+var _ ServerError = driver.WriteCommandError{}
```

Comment:
Would returning a `driver.WriteCommandError` or `driver.WriteConcernError` just be a bug? 

IIUC, `processWriteError` rewrites them in terms of `mongo` errors